taichi/codegen/codegen_opengl.cpp

```diff
@@ -521,11 +524,24 @@ class KernelGen : public IRVisitor {
       emit("int _tid = int(gl_GlobalInvocationID.x);");
       emit("if (_tid >= {}) return;", num_threads_);
       emit("int _itv = {} + _tid * {};", begin_value, 1 /* stmt->step? */);
+      stmt->body->accept(this);
     } else {
-      TI_ERROR("[glsl] non-const range_for currently unsupported under OpenGL");
+
+      ScopedIndent _s(line_appender_);
+      emit("// range known at runtime");
+      num_threads_ = 1; // TODO: grid-stride-loop
+      auto begin_expr = stmt->const_begin ? std::to_string(stmt->begin_value)
+        : fmt::format("_gtmp_i32_[{} >> 2]", stmt->begin_offset);
+      auto end_expr = stmt->const_end ? std::to_string(stmt->end_value)
+        : fmt::format("_gtmp_i32_[{} >> 2]", stmt->end_offset);
+      emit("int _tid = int(gl_GlobalInvocationID.x);");
+      emit("if (_tid >= {}) return;", num_threads_);
```

Comment:
I guess this means for now, only one thread is used to handle all the works?

Reply:
Yes! Should I do grid-stride loop in another PR, or not?

Reply:
Yep, please do it in another PR and keep this one small..